x/dymns/client/cli/query_buy_order.go

```diff
@@ -0,0 +1,238 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/version"
+	dymnsutils "github.com/dymensionxyz/dymension/v3/x/dymns/utils"
+	"github.com/spf13/cobra"
+
+	dymnstypes "github.com/dymensionxyz/dymension/v3/x/dymns/types"
+)
+
+const (
+	flagTargetType = "target-type"
+
+	targetTypeById    = "offer-id"
+	targetTypeBuyer   = "buyer"
+	targetTypeOwner   = "owner"
+	targetTypeDymName = "name"
+	targetTypeAlias   = "alias"
+	targetTypeRollApp = "rollapp"
+)
+
+// CmdQueryBuyOrder is the CLI command for querying Buy-Orders a Dym-Name
+func CmdQueryBuyOrder() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "buy-order [target]",
+		Aliases: []string{"offer"},
+		Short:   "Get list of Buy-Orders corresponding to the target",
+		Example: fmt.Sprintf(
+			`%s q %s offer 1 --%s=%s
+%s q %s offer dym1buyer --%s=%s
```

Comment:
- `offer` is a cmd alias of `buy-order`. I use shorter for the example. Like in `resolve-dym-name-address` cmd, I defined cmd alias `resolve` and use it in example.
- It's a bit confused because this query supports multiple type of queries. I already improved it a bit.